### ISSUE TITLE
Issue with baked commands in 1.12.5

### ISSUE THREAD

**Issue by javierpena** [amoffat/sh#340]
Hi,

We're having an issue with sh 1.12.5 when using baked commands. The following code works fine with 1.11.0 and 1.12.0:

```
import sh

url="https://github.com/openstack/puppet-nova"
path="/tmp/nova"
branch="master"

print("Starting clone %s to %s" % (url, path))
sh.git.clone(url, path)
print("Cloned %s to %s" % (url, path))

git = sh.git.bake(_cwd=path, _tty_out=False, _timeout=3600)
try:
    print("Fetching origin")
    git.fetch("origin")
    print("Fetched origin")
except Exception:
    print("Error fetching into %s, deleting." % (path))
```
However, with 1.12.5 it never comes back from git.fetch("origin"). I see the git process has finished, but the Python call never returns.

**Comment by javierpena** [comment]
I've run some more tests. It worked fine in 1.12.3, and started breaking in 1.12.4

**Comment by amoffat** [comment]
Confirmed.  Thanks for reporting.  I did some minor refactoring in 1.12.4, and apparently the tests did not catch this regression.  Looking into it now.

**Comment by amoffat** [comment]
Temporary workaround seems to be to remove the timeout.

**Comment by amoffat** [comment]
dup of https://github.com/amoffat/sh/issues/340, closing

**Comment by amoffat** [comment]
the issue is fixed on the latest release, 1.12.6
